src/server/rdb_load.cc

```diff
@@ -1004,6 +1004,17 @@ void RdbLoaderBase::OpaqueObjLoader::HandleBlob(string_view blob) {
     }
     pv_->InitRobj(OBJ_ZSET, encoding, inner);
     return;
+  } else if (rdb_type_ == RDB_TYPE_ZSET_LISTPACK) {
+    if (!lpValidateIntegrity((uint8_t*)blob.data(), blob.size(), 0, nullptr, nullptr)) {
+      LOG(ERROR) << "ListPack integrity check failed.";
+      ec_ = RdbError(errc::rdb_file_corrupted);
+      return;
+    }
+    unsigned char* src_lp = (unsigned char*)blob.data();
+    unsigned long long bytes = lpBytes(src_lp);
+    unsigned char* lp = (uint8_t*)zmalloc(bytes);
+    ::memcpy(lp, src_lp, bytes);
```

Comment:
Just curios, why the global namespace? I always thought its better to use the std re-exports of c functions?

Reply:
That was my mistake, I never use this IMO, but I wanted to push this code before I left on Firday and somehow I ended up doing this